lib/ansible/modules/windows/win_dsc.ps1

```diff
@@ -0,0 +1,231 @@
+#!powershell
+# (c) 2015, Trond Hindenes <trond@hindenes.com>, and others
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+#Temporary fix
+#Set-StrictMode -Off
+
+$params = Parse-Args $args -supports_check_mode $true
+$result = New-Object psobject
+Set-Attr $result "changed" $false
+
+#Check that we're on at least Powershell version 5
+if ($PSVersionTable.PSVersion.Major -lt 5)
+{
+    Fail-Json -obj $Result -message "This module only runs on Powershell version 5 or higher"
+}
+
+$CheckMode = $False
+$CheckFlag = $params.psobject.Properties | where {$_.Name -eq "_ansible_check_mode"}
+if ($CheckFlag)
+{
+    if (($CheckFlag.Value) -eq $True)
+    {
+        $CheckMode = $True
+    }
+
+}
+
+$resourcename = Get-Attr -obj $params -name resource_name -failifempty $true -resultobj $result
+
+#From Ansible 2.3 onwards, params is now a Hash Array
+$Attributes = $params.GetEnumerator() | where {$_.key -ne "resource_name"} | where {$_.key -notlike "_ansible_*"}
+
+
+if (!($Attributes))
+{
+    Fail-Json -obj $result -message "No attributes specified"
+}
+
+#Always return the name
+set-attr -obj $result -name "resource_name" -value $resourcename
+set-attr -obj $result -name "attributes" -value $Attributes
+set-attr -obj $result -name "reboot_required" -value $null
+
+# Build Attributes Hashtable for DSC Resource Propertys
+$Attrib = @{}
+foreach ($key in $Attributes)
+{
+    set-attr -obj $result -name $key.name -value $key.value
+    $Attrib.Add($Key.Key,$Key.Value)
+}
+Set-Attr -obj $result -name DSCAttributes = -value $Attrib
+
+<#
+$params.Keys | foreach-object {
+    $Attrib.Add($_,$params.Item($_))
+    set-attr -obj $result -name $_ -value $params.Item($_)
+    }
+#>
+
+$Config = @{
+   Name = ($resourcename)
+   Property = @{
+        }
+    }
+
+
+$Resource = Get-DscResource -Name $resourcename -ErrorAction SilentlyContinue
+if (!$Resource)
+{
+    Fail-Json -obj $result -message "Resource $resourcename not found"
+}
+
+#Get the Module that provides the resource. Will be used as
+#mandatory argument for Invoke-DscResource
+$Module = $Resource.ModuleName
+if (@($Module).Count -gt 1) {
+    Fail-Json -obj $result -message "Multiple DSC modules found with resource name [$($resourcename)]"
```

Comment:
I know, I was never happy about that part. The idea is that if the "called" DSC resource exists in more than one module, I don't know which one to load. However, I think I'll just default to the latest one, as that's the convention Powershell uses all over. I'll refactor it right now.